allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -1579,16 +1643,15 @@ def run_eval(
         ckpt = self.checkpoint_load(checkpoint_file_path)
         total_steps = cast(int, ckpt["total_steps"])
 
-        rollouts = RolloutStorage(
-            num_steps=rollout_steps,
-            num_samplers=self.num_samplers,
-            actor_critic=cast(ActorCriticModel, self.actor_critic),
-        )
+        training_pipeline = self.config.training_pipeline()
```

Comment:
I agree, it hurt me a bit when I had instantiate the training pipeline here. I think the "right" thing to do is to have an `eval_pipeline` (or maybe even just a `eval_rollout_storage` as that's all we're taking from the training pipeline) but I'm hesitant to require users to specify yet another thing as I think we require people to implement quite a few functions already... I'd be happy to hear a proposal though.

Reply:
I might be missing an important detail, but since for eval we only use the storage for the model's recurrent memory (right?), I think we could just shortcut the entire cap to `training_pipeline` by just doing `rollout_storage=RolloutBlockStorage(init_size=rollout_steps)` (adding `from allenact.algorithms.onpolicy_sync.storage import RolloutBlockStorage`)...